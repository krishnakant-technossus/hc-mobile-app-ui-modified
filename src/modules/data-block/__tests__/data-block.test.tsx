// npx jest ./src/modules/data-block/__tests__/data-block.test.ts

import * as React from 'react';
import {render} from '@testing-library/react-native';

import DataBlock from '../data-block';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

const dummyItems = [
  {
    value: 'Value One',
    label: 'Label One',
  },
  {
    value: 'Value Two',
    label: 'Label Two',
  },
];

describe('DataBlock', () => {
  describe('Rendering', () => {
    it('should render properly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <DataBlock items={dummyItems} />
        </ThemeProvider>,
      );

      const labelOne = wrapper.getByText(dummyItems[0].label);
      const labelTwo = wrapper.getByText(dummyItems[1].label);
      const valueOne = wrapper.getByText(dummyItems[0].value);
      const valueTwo = wrapper.getByText(dummyItems[1].value);

      expect(labelOne).toBeDefined();
      expect(labelTwo).toBeDefined();
      expect(valueOne).toBeDefined();
      expect(valueTwo).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });

    it('should render null when items are not passed', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <DataBlock />
        </ThemeProvider>,
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
