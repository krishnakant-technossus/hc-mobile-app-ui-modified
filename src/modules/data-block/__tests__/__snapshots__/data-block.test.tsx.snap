// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataBlock Rendering should render null when items are not passed 1`] = `null`;

exports[`DataBlock Rendering should render properly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomColor": "#DBE0E3",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 15,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flexBasis": 0,
            "flexGrow": 0.6,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#1C2530",
              "fontFamily": "System",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "textAlign": "left",
            },
          ]
        }
        testID="text"
        variant="caption"
      >
        Value One
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "flexBasis": 0,
            "flexGrow": 0.4,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        muted={true}
        style={
          Array [
            Object {
              "color": "#97A4AB",
              "fontFamily": "System",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "textAlign": "left",
            },
          ]
        }
        testID="text"
      >
        Label One
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomColor": "#DBE0E3",
          "borderBottomWidth": 0,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 15,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flexBasis": 0,
            "flexGrow": 0.6,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#1C2530",
              "fontFamily": "System",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "textAlign": "left",
            },
          ]
        }
        testID="text"
        variant="caption"
      >
        Value Two
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "flexBasis": 0,
            "flexGrow": 0.4,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        muted={true}
        style={
          Array [
            Object {
              "color": "#97A4AB",
              "fontFamily": "System",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "textAlign": "left",
            },
          ]
        }
        testID="text"
      >
        Label Two
      </Text>
    </View>
  </View>
</View>
`;
